osu.Framework/Graphics/Containers/SearchContainer.cs

```diff
@@ -121,7 +121,7 @@ protected void Filter()
         private void performFilter()
         {
             string[] terms = (searchTerm ?? string.Empty).Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
To avoid the unnecessary allocation. Not related to this PR, but noticed it while reading through it. Wanna include it in this PR while you are at it?
I'm surprised that there is no analyzer for this 😅

```suggestion
            string[] terms = (searchTerm ?? string.Empty).Split(' ', StringSplitOptions.RemoveEmptyEntries);
```

Reply:
Yeah surprisingly, there isn't a single rule for switching from collection parameter to singleton. CA1861 is about using collections more efficiently.
I'd like to avoid including unrelated changes for this PR, because the discovered instance numbers can be non-trivial.